gadget/install/encrypt_test.go

```diff
@@ -58,78 +58,123 @@ func (s *encryptSuite) SetUpTest(c *C) {
+			formatErr: nil,
+			openErr:   "",
+			err:       "",
 		},
 		{
-			"cryptsetup", "open", "--key-file", "-", "/dev/node1", "some-label",
+			formatErr: errors.New("format error"),
+			openErr:   "",
+			err:       "cannot format encrypted device: format error",
 		},
-	})
-
-	err = dev.Close()
-	c.Assert(err, IsNil)
-}
-
-func (s *encryptSuite) TestEncryptFormatError(c *C) {
-	s.mockCryptsetup = testutil.MockCommand(c, "cryptsetup", `[ "$2" == "luksFormat" ] && exit 127 || exit 0`)
-	s.AddCleanup(s.mockCryptsetup.Restore)
-
-	key := secboot.EncryptionKey{}
-	_, err := install.NewEncryptedDevice(&mockDeviceStructure, key, "some-label")
-	c.Assert(err, ErrorMatches, "cannot format encrypted device:.*")
-}
-
-func (s *encryptSuite) TestEncryptOpenError(c *C) {
-	s.mockCryptsetup = testutil.MockCommand(c, "cryptsetup", `[ "$1" == "open" ] && exit 127 || exit 0`)
-	s.AddCleanup(s.mockCryptsetup.Restore)
-
-	key := secboot.EncryptionKey{}
-	_, err := install.NewEncryptedDevice(&mockDeviceStructure, key, "some-label")
-	c.Assert(err, ErrorMatches, "cannot open encrypted device on /dev/node1:.*")
-}
-
-func (s *encryptSuite) TestEncryptAddKey(c *C) {
-	capturedFifo := filepath.Join(c.MkDir(), "captured-stdin")
-	s.mockCryptsetup = testutil.MockCommand(c, "cryptsetup", fmt.Sprintf(`[ "$1" == "luksAddKey" ] && cat %s/tmp-rkey > %s || exit 0`, dirs.SnapRunDir, capturedFifo))
-	s.AddCleanup(s.mockCryptsetup.Restore)
-
-	key := secboot.EncryptionKey{}
-	dev, err := install.NewEncryptedDevice(&mockDeviceStructure, key, "some-label")
-	c.Assert(err, IsNil)
-
-	rkey := secboot.RecoveryKey{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}
-	err = dev.AddRecoveryKey(key, rkey)
-	c.Assert(err, IsNil)
-
-	c.Assert(s.mockCryptsetup.Calls(), DeepEquals, [][]string{
 		{
-			"cryptsetup", "-q", "luksFormat", "--type", "luks2", "--key-file", "-",
-			"--cipher", "aes-xts-plain64", "--key-size", "512", "--pbkdf", "argon2i",
-			"--iter-time", "1", "--label", "some-label-enc", "/dev/node1",
+			formatErr: nil,
+			openErr:   "open error",
+			err:       "cannot open encrypted device on /dev/node1: open error",
 		},
-		{
-			"cryptsetup", "open", "--key-file", "-", "/dev/node1", "some-label",
-		},
-		{
-			"cryptsetup", "luksAddKey", "/dev/node1", "-q", "--key-file", "-",
-			"--key-slot", "1", filepath.Join(dirs.SnapRunDir, "tmp-rkey"),
-		},
-	})
-	c.Assert(capturedFifo, testutil.FileEquals, rkey[:])
+	} {
+		script := ""
+		if tc.openErr != "" {
+			script = fmt.Sprintf("echo '%s'>&2; exit 1", tc.openErr)
+
+		}
+		s.mockCryptsetup = testutil.MockCommand(c, "cryptsetup", script)
+		s.AddCleanup(s.mockCryptsetup.Restore)
+
+		// create empty key to prevent blocking on lack of system entropy
+		myKey := secboot.EncryptionKey{}
+		for i := range myKey {
+			myKey[i] = byte(i)
+		}
+
+		calls := 0
+		restore := install.MockSecbootFormatEncryptedDevice(func(key secboot.EncryptionKey, label, node string) error {
```

Comment:
We are not checking node here, why is that?

Reply:
Oops, this was really missing, thanks Michael.